sync/sdk.md

```diff
@@ -0,0 +1,100 @@
+# Sync SDK
+
+## Platforms
+
+The initial targeted platforms for a Sync SDK are:
+
+* Android
+* iOS
+* Cordova
+
+Progressive Web App (PWA) may be targetted at a later stage.
+
+## MemeoList App for 'dogfooding'
+
+During development of the Data Sync Server and the Client SDKs, it will be helpful to have Apps with real functionality and use cases.
+2 Apps were considered for this:
+
+* Memeolist App (more info at https://github.com/aerogear/proposals/pull/41)
+* Cool Stuff Store App
+
+At this time, only the Memeolist App will be used for this 'dogfooding' purpose. However, it may make sense to convert/write other Apps to work with Data Sync. 
+
+## Binary upload
+
+Binary upload in the context of GraphQL means supporting the upload of binary content from an App to a Mutation resolver.
```

Comment:
Instead of send a _Blob_ why not convert it into a byte[]/String on the client side?

Reply:
Do you mean a base64 string?
While that will work, it sounds like a temporary solution/hack.
Ideally binary content (like an image) is transferred as binary to avoid unnecessary overhead, particularly if there's potential for network connectivity issues on mobile.

However, if the solution for supporting binary is too complex or not worth the effort, base64 may be the best solution

Reply:
We will probably want to separate the "sync" object from large binary objects.  This way we can quickly populate client side graphs without needing a lot of handling for updates failing because someone attached a 100MB zip file.  Of course with GraphQL this is easy because we can have a "nobinary" query that sync talks to and a "details" query that pulls down the large data of a single object.

For uploads, we want to avoid copying large data sets into RAM and uploading those, aside from that I don't have strong opinions of how things are uploaded.